filecrypt/sops.go

```diff
@@ -0,0 +1,65 @@
+package filecrypt
+
+import (
+	"github.com/Ibotta/sopstool/fileutil"
+	"github.com/Ibotta/sopstool/oswrap"
+)
+
+type sopsCrypt struct {
+	execWrap oswrap.ExecWrap
+	osWrap   oswrap.OsWrap
+}
+
+var sops = sopsCrypt{
+	execWrap: oswrap.ExecWrapInstance(),
+	osWrap:   oswrap.OsWrapInstance(),
+}
+
+// SopsCryptInstance gets an instance of the sops wrapper
+func SopsCryptInstance() FileCrypt {
+	return sops
+}
+
+// EncryptFile encrypts a file rewriting the sops encrypted file
+func (sops sopsCrypt) EncryptFile(fn string) error {
+	cryptfile := fileutil.NormalizeToSopsFile(fn)
+	return sops.execWrap.RunCommandStdoutToFile(cryptfile, []string{"sops", "-e", fn})
+}
+
+// DecryptFile decrypts a file rewriting the plaintext file
+func (sops sopsCrypt) DecryptFile(fn string) error {
+	cryptfile := fileutil.NormalizeToSopsFile(fn)
+	return sops.execWrap.RunCommandStdoutToFile(fn, []string{"sops", "-d", cryptfile})
+}
+
+// DecryptFilePrint decrypts a file printing the result
+func (sops sopsCrypt) DecryptFilePrint(fn string) error {
+	cryptfile := fileutil.NormalizeToSopsFile(fn)
+	return sops.execWrap.RunCommandDirect([]string{"sops", "-d", cryptfile})
+}
+
+// RemoveFile removes a plaintext file from the filesystem
+func (sops sopsCrypt) RemoveFile(fn string) error {
+	return sops.osWrap.Remove(fn)
```

Comment:
This error if the file does not exist.  Which I think is the normal scenario this would be used.  If I clone a repo with secrets, try to remove using `-d` the plain text version will not exist on disk and cause this to error.

Reply:
Thanks. I think this is captured in https://github.com/Ibotta/sopstool/issues/4